src/mame/casio/cz230s.cpp

```diff
@@ -0,0 +1,693 @@
+		if (BIT(m_key_mux, 0))
+			data &= (input & 0x3f);
+		if (BIT(m_key_mux, 1))
+			data &= (input >> 6);
+	}
+
+	return data;
+}
+
+/**************************************************************************/
+template <int Row>
+CUSTOM_INPUT_MEMBER(cz230s_state::keys_row_r)
+{
+	u8 data = 0xff;
+
+	for (int i = 0; i < 5; i++)
+		if (BIT(m_key_sel, i))
+			data &= m_keys[i].read_safe(0xff);
+
+	return BIT(data, Row);
+}
+
+/**************************************************************************/
+template <int Row>
+u8 cz230s_state::keys_analog_r()
+{
+	return keys_row_r<Row>() ? 0xff : 0x00;
+}
+
+/**************************************************************************/
+INPUT_CHANGED_MEMBER(cz230s_state::rhythm_w)
+{
+	if (!oldval && newval)
+		m_rhythm = param;
+}
+
+/**************************************************************************/
+INPUT_CHANGED_MEMBER(cz230s_state::mode_w)
+{
+	if (!oldval && newval)
+		m_mode = param;
+}
+
+/**************************************************************************/
+void cz230s_state::cassette_w(int state)
+{
+	m_cassette->output(state ? -1.0 : 1.0);
+}
+
+/**************************************************************************/
+void cz230s_state::cassette_motor_w(int state)
+{
+	m_cassette->change_state(state ? CASSETTE_MOTOR_ENABLED : CASSETTE_MOTOR_DISABLED, CASSETTE_MASK_MOTOR);
+}
+
+/**************************************************************************/
+CUSTOM_INPUT_MEMBER(cz230s_state::cassette_r)
+{
+	return m_cassette->input() > 0 ? 0 : 1;
+}
+
+/**************************************************************************/
+void cz230s_state::render_w(int state)
+{
+	if(!state)
+		return;
+
+	for (int digit = 0; digit < 6; digit++)
+	{
+		const u16 data = m_lcdc->output(digit);
+		for (int seg = 0; seg < 9; seg++)
+			m_lcd_seg[digit][seg] = BIT(data, seg);
+	}
+}
+
+
+//**************************************************************************
+//  MACHINE DEFINTIONS
+//**************************************************************************
+
+void cz230s_state::config_base(machine_config &config, u16 screen_w, u16 screen_h, bool midi_thru)
+{
+	UPD7811(config, m_maincpu, 10_MHz_XTAL);
+
+	CLOCK(config, "midi_clock", 2_MHz_XTAL).signal_handler().set(m_maincpu, FUNC(upd7810_device::sck_w));
+
+	midi_port_device &mdin(MIDI_PORT(config, "mdin", midiin_slot, "midiin"));
+	mdin.rxd_handler().set([this](int state) { m_midi_rx = state; });
+	m_maincpu->rxd_func().set([this]() { return m_midi_rx; });
+
+	MIDI_PORT(config, "mdout", midiout_slot, "midiout");
+	m_maincpu->txd_func().set("mdout", FUNC(midi_port_device::write_txd));
+
+	if (midi_thru)
+	{
+		MIDI_PORT(config, "mdthru", midiout_slot, "midiout");
+		mdin.rxd_handler().append("mdthru", FUNC(midi_port_device::write_txd));
+	}
+
+	NVRAM(config, "nvram", nvram_device::DEFAULT_ALL_0);
+
+	MN1252(config, m_lcdc);
+
+	auto& screen = SCREEN(config, "screen", SCREEN_TYPE_SVG);
+	screen.set_refresh_hz(60);
```

Comment:
This right-associative unary `&` is on the left of the space, unlike the other ones which are on the right of the space.